### PR TITLE
da1469x disable dcdc when vbus is present

### PR DESCRIPTION
**Pull request by kasjer**
DCDC is more efficient then LDO to provide power rails but in case
when VBUS (charger) is present, it is actually more efficient from
battery point of view to disable DCDC (which is powered by battery)
and use LDO that is powered by VBUS.
This way charging current will go to battery and not to DCDC reducing charge time.
It will also prevent discharging battery when it was fully changed and not removed
from charger.

### CODE REVIEW

File: hw/mcu/dialog/da1469x/src/da1469x_prail.c [hunk 1]
```diff
@@ -24,6 +24,9 @@
 #include "mcu/da1469x_hal.h"
 #include "mcu/da1469x_prail.h"
 #include "mcu/da1469x_retreg.h"
+#if MYNEWT_VAL(ENABLE_BLE)
```
Comment (andrzej-kaczmarek) [inline hw/mcu/dialog/da1469x/src/da1469x_prail.c:27]: there's no such setting, use `MYNEWT_VAL_CHOICE(BLE_HCI_TRANSPORT, dialog_cmac)`

File: hw/mcu/dialog/da1469x/src/da1469x_prail.c [hunk 2]
```diff
@@ -176,13 +179,21 @@ da1469x_prail_dcdc_restore(void)
     if (CRG_TOP->ANA_STATUS_REG & CRG_TOP_ANA_STATUS_REG_COMP_VBAT_HIGH_Msk) {
         da1469x_retreg_restore(g_mcu_dcdc_config, ARRAY_SIZE(g_mcu_dcdc_config));
         DCDC->DCDC_CTRL1_REG |= DCDC_DCDC_CTRL1_REG_DCDC_ENABLE_Msk;
+#if MYNEWT_VAL(ENABLE_BLE)
+        /* Enable turnning DCDC on from CMAC core. */
```
Comment (andrzej-kaczmarek) [inline hw/mcu/dialog/da1469x/src/da1469x_prail.c:183]: typo

File: hw/mcu/dialog/da1469x/src/da1469x_prail.c [hunk 3]
```diff
@@ -176,13 +179,21 @@ da1469x_prail_dcdc_restore(void)
...
+#endif
     }
 }
 #endif
 
 void
 da1469x_prail_dcdc_disable(void)
 {
+#if MYNEWT_VAL(ENABLE_BLE)
+    /* Prevent CMAC from turnning DCDC on. */
```
Comment (andrzej-kaczmarek) [inline hw/mcu/dialog/da1469x/src/da1469x_prail.c:194]: typo

File: hw/drivers/chg_ctrl/da1469x_charger/src/da1469x_charger.c [hunk 4]
```diff
@@ -27,6 +27,8 @@
 #include <bsp/bsp.h>
 #if MYNEWT_VAL(DA1469X_CHARGER_USE_CHARGE_CONTROL)
 #include <charge-control/charge_control.h>
+#include <mcu/da1469x_prail.h>
+
```
Comment (andrzej-kaczmarek) [inline hw/drivers/chg_ctrl/da1469x_charger/src/da1469x_charger.c:31]: empty line

**Review by andrzej-kaczmarek — CHANGES_REQUESTED** [review]

File: hw/mcu/dialog/da1469x/src/da1469x_prail.c (hunk 1 above)
Comment (kasjer) [reply hw/mcu/dialog/da1469x/src/da1469x_prail.c]: done

File: hw/mcu/dialog/da1469x/src/da1469x_prail.c (hunk 2 above)
Comment (kasjer) [reply hw/mcu/dialog/da1469x/src/da1469x_prail.c]: done

File: hw/drivers/chg_ctrl/da1469x_charger/src/da1469x_charger.c (hunk 4 above)
Comment (kasjer) [reply hw/drivers/chg_ctrl/da1469x_charger/src/da1469x_charger.c]: done

**Review by andrzej-kaczmarek — APPROVED** [review]

**Pull request merged**